pkg/controller/argocd/applicationset.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2021 ArgoCD Operator Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package argocd
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	argoprojv1a1 "github.com/argoproj-labs/argocd-operator/pkg/apis/argoproj/v1alpha1"
+	"github.com/argoproj-labs/argocd-operator/pkg/controller/argoutil"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *ReconcileArgoCD) reconcileApplicationSetController(cr *argoprojv1a1.ArgoCD) error {
+
+	log.Info("reconciling applicationset serviceaccounts")
+	sa, err := r.reconcileApplicationSetServiceAccount(cr)
+	if err != nil {
+		return err
+	}
+
+	log.Info("reconciling applicationset roles")
+	role, err := r.reconcileApplicationSetRole(cr)
+	if err != nil {
+		return err
+	}
+
+	log.Info("reconciling applicationset role bindings")
+	if err := r.reconcileApplicationSetRoleBinding(cr, role, sa); err != nil {
+		return err
+	}
+
+	log.Info("reconciling applicationset deployments")
+	if err := r.reconcileApplicationSetDeployment(cr, sa); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// reconcileApplicationControllerDeployment will ensure the Deployment resource is present for the ArgoCD Application Controller component.
+func (r *ReconcileArgoCD) reconcileApplicationSetDeployment(cr *argoprojv1a1.ArgoCD, sa *corev1.ServiceAccount) error {
+	deploy := newDeploymentWithSuffix("applicationset-controller", "controller", cr)
+
+	setAppSetLabels(&deploy.ObjectMeta)
+
+	podSpec := &deploy.Spec.Template.Spec
+
+	podSpec.ServiceAccountName = sa.ObjectMeta.Name
+
+	podSpec.Containers = []corev1.Container{{
+		Command: []string{"applicationset-controller", "--argocd-repo-server", generateRepoServerAddress(cr)},
+		Env: []corev1.EnvVar{{
+			Name: "NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		}},
+		Image:           "quay.io/argocdapplicationset/argocd-applicationset:v0.1.0",
```

Comment:
Please follow the pattern in deployment.go/getArgoContainerImage() to allow default image to be changed by env var.  Also, please update the docs/reference/argocd.md to include new env var.

Reply:
Thanks @wtam2018 , done and done, let me know if this is what you were looking for...